compiler/resolution/implementForallIntents2.cpp

```diff
@@ -0,0 +1,1097 @@
+/*
+ * Copyright 2004-2017 Cray Inc.
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "implementForallIntents.h"
+
+/*
+This file contains Part 2 of the new implementation of forall intents.
+"Part 2" is mostly about extending a copy of the parallel iterator in question
+with arguments and yield values corresponding to forall intents.
+
+For a reduce intent, the global reduce op instance needs to be
+properly allocated, deallocated and connected to the corresponding
+user outer variable. This is the job of "Part 1".
+
+For now, implementForallIntents2New() and immediately-adjacent functions
+are in implementForallIntents.cpp, for historical reasons.
```

Comment:
Eh... why should you be conservative about moving code between the two files? If they fit better here, move them. My one "it would have been nice" request about this is that this PR seems to have a single commit combining code changes and code movement. That makes it much harder to follow. Try to move code in a separate commit.